frontend/src/services/settings.ts

```diff
@@ -14,6 +16,29 @@ export const DEFAULT_SETTINGS: Settings = {
 
 const validKeys = Object.keys(DEFAULT_SETTINGS) as (keyof Settings)[];
 
+export const getCurrentSettingsVersion = () => {
+  const settingsVersion = localStorage.getItem("SETTINGS_VERSION");
+  if (!settingsVersion) return 0;
+  try {
+    return parseInt(settingsVersion, 10);
+  } catch (e) {
+    return 0;
+  }
+};
+
+export const settingsAreUpToDate = () => (
+  getCurrentSettingsVersion() === LATEST_SETTINGS_VERSION
+);
+
+export const maybeMigrateSettings = () => {
+  // Sometimes we ship major changes, like a new default agent.
+  // In this case, we may want to override a previous choice made by the user.
+  const currentVersion = getCurrentSettingsVersion();
+  if (currentVersion < 1) {
```

Comment:
Don't you mean `if (currentVersion < LATEST_SETTINGS_VERSION)`?

Reply:
Nope!

This is a migration to get to version 1. When version 2 comes out, most users will have already been pushed into CodeActAgent, and some might have deliberately chosen a different agent.

I expect this function to grow into a big set if `if (currentVersion < x)`, where `x` will get added each time the version increments to `x`